services/alertmanager/alertmanager.go

```diff
@@ -157,6 +169,159 @@ func (svc *Service) updateConfiguration(ctx context.Context) {
 	svc.l.Infof("%s created", alertmanagerConfigPath)
 }
 
+func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
+	var settings *models.Settings
+	var rules []*models.Rule
+	var channels []*models.Channel
+	e := svc.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		settings, err = models.GetSettings(tx.Querier)
+		if err != nil {
+			return err
+		}
+
+		rules, err = models.FindRules(tx.Querier)
+		if err != nil {
+			return err
+		}
+
+		channels, err = models.FindChannels(tx.Querier)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if e != nil {
+		return errors.Errorf("Failed to fetch items from database: %s", e)
+	}
+
+	cfg.Global = &alertmanager.GlobalConfig{}
```

Comment:
It looks like you should allocte new config only if `cfg.Global` is nil, otherwise you will lose user defined values.